src/Statement/StatementListChanger.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Statement;
+
+use Wikibase\DataModel\Entity\PropertyId;
+use Wikibase\DataModel\Statement\Statement;
+use Wikibase\DataModel\Statement\StatementList;
+
+/**
+ * A collection of functions to manipulate StatementList objects.
+ *
+ * @license GPL-2.0+
+ * @author Thiemo Mättig
+ */
+class StatementListChanger {
```

Comment:
We already have a `StatementGrouper` interface with multiple implementations (including a `ByPropertyIdStatementGrouper`), as well as a `ByPropertyIdArray` (that's what we need to get rid of) and a `ByPropertyIdGrouper`. I wanted this here to be different from these.

This is a layer of code that works with a StatementList while being aware of the concept of "grouping by property ID". ByPropertyIdAwareStatementListChanger?